stix_shifter/stix_translation/src/modules/azure_sentinel/query_constructor.py

```diff
@@ -0,0 +1,406 @@
+            # Resolve the comparison symbol to use in the query string (usually just ':')
+            comparator = self._lookup_comparison_operator(expression.comparator)
+
+            # Some values are formatted differently based on how they're being compared
+            if expression.comparator == ComparisonComparators.Matches:  # needs forward slashes
+                value = self._format_match(expression.value)
+            # should be (x, y, z, ...)
+            elif expression.comparator == ComparisonComparators.In:
+                value = self._format_set(expression.value)
+            elif expression.comparator == ComparisonComparators.Equal \
+                    or expression.comparator == ComparisonComparators.NotEqual \
+                    or expression.comparator == ComparisonComparators.GreaterThan \
+                    or expression.comparator == ComparisonComparators.LessThan \
+                    or expression.comparator == ComparisonComparators.GreaterThanOrEqual \
+                    or expression.comparator == ComparisonComparators.LessThanOrEqual:
+                # Should be in single-quotes
+                value = self._format_equality(expression.value)
+            # '%' -> '*' wildcard, '_' -> '?' single wildcard
+            elif expression.comparator == ComparisonComparators.Like:
+                value = self._format_like(expression.value)
+            else:
+                value = self._escape_value(expression.value)
+
+            if expression.negated:
+                if expression.comparator in [ComparisonComparators.Like, ComparisonComparators.Matches]:
+                    raise SearchFeatureNotSupportedError("'NOT' Operator is not supported for LIKE and MATCHES")
+                elif stix_object in ['ipv4-addr', 'ipv6-addr'] or stix_field in ['src_ref.value', 'dst_ref.value']:
+                    raise SearchFeatureNotSupportedError("'NOT' Operator is not supported for IPV4 or IPV6 address")
+                comparator = self.negated_comparator_lookup.get(expression.comparator)
+
+            # to remove single quotes in specific field value
+            if stix_field in ['pid', 'parent_ref.pid', 'account_last_login']:
+                if expression.comparator in [ComparisonComparators.Like, ComparisonComparators.Matches]:
+                    raise SearchFeatureNotSupportedError('"{operator}" operator is not supported for '
+                                                         '"{stix_field}" attribute'
+                                                         .format(operator=expression.comparator.name.upper(),
+                                                                 stix_field=stix_field))
+                value = self._format_value_without_quotes(value)
+
+            if stix_field not in ['provider', 'vendor']:
+                value = self._format_value_to_lower_case(value)
+
+            self.COUNTER += 1
```

Comment:
some comments would be helpful for this COUNTER global variable

Reply:
Comment line updated for the COUNTER variable. COUNTER is used to form sequential lambda function names for OData4 queries per comparison observation.
e.g. processes/any(query1:contains(tolower(query1/path), 'c:\\windows\\system32')) and processes/any(query2:contains(tolower(query2/name), 'exe')) '''. Here 'query1' and 'query2' are lambda function names used in the OData queries for better readability and query execution.